Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp

```diff
@@ -669,15 +770,6 @@ namespace AZ::SettingsRegistryMergeUtils
         if ([[maybe_unused]] constexpr auto projectPathKey = FixedValueString(BootstrapSettingsRootKey) + "/project_path";
             !projectPath.empty())
         {
-            if (projectPath.IsRelative())
```

Comment:
just a question: I see you move path conversion from this function to engine&project path getter function, why do we need to do it?

Reply:
Because these functions are called outside of SettingsRegistryMergeUtils.cpp such as in the ProjectManager.cpp [FindProjectAndEngineRootPaths](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/ProjectManager/ProjectManager.cpp#L50-L51) to retrieve the project path and those paths need to absolute.
Otherwise if a relative path is returned, when the [ProjectManager::CheckProjectPathProvided](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/ProjectManager/ProjectManager.cpp#L67) function checks if the project path contains a project.json file, then it would append the relative project path to the engine root path instead of the current working directory.

This is an issue when launching the Editor with a relative project path

For example on linux a user has project located at `/home/user/o3de/AutomatedTesting` and the engine located at `/home/user/o3de`

```
[/home/user/o3de/build/linux_ninja/bin/profile] user$ Editor --project-path ../../../../AutomatedTesting
```

The `CheckProjectPathProvided` function would attempt to check for the AutomatedTesting project `project.json` file at <engine-root>/<project-path>/project.json which results in `/home/user/o3de/../../../../AutomatedTesting/project.json`, which really resolves to looking in the Linux root directory for a path of `/Automated/project.json` which fails.

When the project path is absolute, the path append rules has any path segment that is an absolute path replace all the previous path [segments](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L398-L421) with itself.

So if the relative path of `./../../../AutomatedTesting` was resolved to an absolute path, it would become `/home/user/o3de/AutomatedTesting`

Afterwards when the path append operation occurs between the engine root and the project path in [CheckProjectPathProvided](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/ProjectManager/ProjectManager.cpp#L67) function it would append the paths of
```
"/home/user/o3de" / "/home/user/o3de/AutomatedTesting" / "project.json"
```
Because the project path in that list is absolute, the engine path segment would be discarded resulting in the correct path of "/home/user/o3de/AutomatedTesting/project.json".

Our path library behavior is consistent with the Python pathlib and [os.path.json](https://docs.python.org/3/library/os.path.html#os.path.join) behavior
> If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.